### DISCUSSION TITLE
Does alembic have repeatable migrations ?

### DISCUSSION THREAD

**Discussion by gerazenobi** [sqlalchemy/alembic#950] [Usage Questions] [answered]
Something like what liquibase or flyway provide:
https://flywaydb.org/documentation/concepts/migrations.html#repeatable-migrations

**Comment by zzzeek** [comment, marked as answer]
this is not a first class feature of Alembic but you can certainly make yourself a set of Python files that will invoke a series of "CREATE OR REPLACE" statements after migrations are updated.  an approach towards SPs, triggers etc that's something like this is described at https://alembic.sqlalchemy.org/en/latest/cookbook.html#replaceable-objects.  A third party library [alembic-utils](https://github.com/olirice/alembic_utils) provides a full , supported version of this pattern.

**Reply by gerazenobi** [reply]
Thanks for the quick reply @zzzeek
